# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gmp/gcc34.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


The resulting macros hacks are to ugly. If anyone needs this optimization
he is free to fix the ugly C hacks (damn when start people learning s.th.
other than this 70's C style ...)

  - Rene Rebe <rene@exactcode.de>

--- gmp-4.1.2/gmp-impl.h.orig	2004-05-05 03:37:08.000000000 +0200
+++ gmp-4.1.2/gmp-impl.h	2004-05-05 03:37:30.000000000 +0200
@@ -577,7 +577,7 @@
    still use this, if we cared to differentiate locals and globals.  */
 
 #if HAVE_HOST_CPU_FAMILY_x86 && __GMP_GNUC_PREREQ (2,96) && ! defined (PIC)
-#define USE_LEADING_REGPARM 1
+#define USE_LEADING_REGPARM 0 /* 1 */
 #else
 #define USE_LEADING_REGPARM 0
 #endif
